yt/visualization/fixed_resolution.py

```diff
@@ -708,21 +712,37 @@ def __getitem__(self, item):
             deposition,
         )
 
-        bounds = []
-        for b in self.bounds:
-            if hasattr(b, "in_units"):
-                b = float(b.in_units("code_length"))
-            bounds.append(b)
+        dd = self.data_source.dd
 
         ftype = item[0]
-        x_data = self.data_source.dd[ftype, self.x_field]
-        y_data = self.data_source.dd[ftype, self.y_field]
-        data = self.data_source.dd[item]
+        if self.axis is None:
+            wd = tuple(el.in_units("code_length").v for el in self.data_source.width)
+            x_data, y_data, rbx0, rbx1, rby0, rby1 = rotate_particle_coord(
+                dd[ftype, "particle_position_x"].to_value("code_length"),
+                dd[ftype, "particle_position_y"].to_value("code_length"),
+                dd[ftype, "particle_position_z"].to_value("code_length"),
+                self.data_source.center.to("code_length").d,
+                wd,
+                self.data_source.normal_vector,
+                self.data_source.north_vector,
+            )
+            bounds = [rbx0, rbx1, rby0, rby1]
+            x_data = np.array(x_data)
+            y_data = np.array(y_data)
+        else:
+            bounds = []
+            for b in self.bounds:
+                if hasattr(b, "in_units"):
```

Comment:
We have an `_ensure_code` in cython, but not in Python.  (We have one that's specific to volume rendering.)